courses/api.py

```diff
@@ -776,48 +777,59 @@ def generate_course_run_certificates():
         return
 
     for run in course_runs:
-        edx_grade_user_iter = exception_logging_generator(
-            get_edx_grades_with_users(run)
-        )
-        created_grades_count, updated_grades_count, generated_certificates_count = (
-            0,
-            0,
-            0,
+        generate_course_run_certificates_for_course(run)
+
+
+def generate_course_run_certificates_for_course(run):
+    """
+    For given course run syncs grades and
```

Comment:
I would update this docstring to specify that the course certificate is generated if the `certificate_available_date` is in the past, OR the CourseRun has `self_paced = True`